### ISSUE TITLE
[Not Compatible] ZSH detected but using sh complete command

### ISSUE THREAD

**Issue by optimizasean** [nanovms/ops#1165]
Running macOS + Homebrew for testing ops.  Installed ops (Version 0.1.26), commands all work fine (so far).

**Note: Nanos Version is 0.0?**

Every time I open a terminal I get this: `.ops/scripts/bash_completion.sh:29: command not found: complete`

It appears that although this recognizes my shell and creates the entries in my .zshrc:
```sh
# OPS config
export OPS_DIR="$HOME/.ops"
export PATH="$HOME/.ops/bin:$PATH"
source "$HOME/.ops/scripts/bash_completion.sh"
```
It fails to do it properly as the file bash_completion.sh uses the command `complete` which is only available in sh, not zsh.

How do I fix this and get rid of the message?  In zsh you should be able to redirect the completions, especially since I am using macOS + brew to here:
```sh
if type brew &>/dev/null; then
    FPATH=$(brew --prefix)/share/zsh-completions:$FPATH

    autoload -Uz compinit
    compinit
fi
```

Potential solutions:
1. Redirect completions to zsh-completions file if brew is detected during install
2. Use autoload to bring in compinit and define autocomplete through compdef on zsh detection during install and add statement to .zshrc/.zprofile if zsh-completions file is not present
3. Change `complete` command to use `autoload` command instead if zsh is detected
4. Ignore the error
5. Delete the autocomplete line on bash_completion.sh:29 so it doesn't appear anymore

Thoughts?

TLDR: I am not currently getting autocomplete on `ops` commands - due to bash_completion.sh using `complete` which is not present in zsh although it does know I am using zsh (because it created .zshrc and added `ops` things into it).

**Comment by eyberg** [comment]
if you installed from source you'll need to do an 'ops update' or 'ops run -n' to get the latest version of nanos (it should install by default through the curl install)

i manually sourced the bash_completion in zsh and seems to work although i might not be using the newer compinit;

can you try adding the following lines before the source line to see if that helps?

```
autoload bashcompinit
bashcompinit
```

if so we could modify the install script for this https://github.com/nanovms/ops/blob/master/install.sh#L280

**Comment by optimizasean** [comment]
Worked, but not sure if the syntax is correct...

```zsh
# zsh-autocompletion
if type brew &>/dev/null; then
    FPATH=$(brew --prefix)/share/zsh-completions:$FPATH
fi

if [ "$ZSH_VERSION" ]; then
    autoload -Uz compinit
    compinit
    autoload bashcompinit
    bashcompinit
fi

# Bash aliases
if [ -f ~/.bash_aliases ]; then
    . ~/.bash_aliases
fi

...in between junk...

# OPS config
export OPS_DIR="$HOME/.ops"
export PATH="$HOME/.ops/bin:$PATH"
source "$HOME/.ops/scripts/bash_completion.sh"
```

This does it but what flags should I be throwing the `autoload` for `bashcompinit` function?  For sure `bashcompinit` must be called after `compinit`, but I am pretty sure that `bashcompinit` is called automatically after everything else loads, so I am not sure if we should be forcefully calling it ourselves in the `.zshrc`?

*Note: I am new to zsh so not too familiar with the inner workings and standards and good practices yet.*

**Comment by eyberg** [comment]
i removed the explicit call to bashcompinit as you are right that it is called after - https://github.com/nanovms/ops/pull/1169 - there are a few patterns used - one is to load it but not execute unless it wasn't defined already (for example another script could've already executed it)

```
autoload +X bashcompinit && bashcompinit
```

this might be more preferable but as long as it's working for you we can prob. merge

**Comment by optimizasean** [comment]
Seems to work fine, no issues on my end that I can tell.

- Updated
- Reinstalled from scratch
- Basic usage
- Usage with more advanced configs

Tested the above.  I would provide the configs but they reveal a bit about the system I am on so I can't share them unfortunately.  Works across the board now for zsh.
